### PR TITLE
[feat] Import HAR API Requests into Workspace

### PR DESCRIPTION
**Pull request by opxdelwin**
This PR addresses the issue #122.

The developed code allows parsing of HAR files, enabling users to choose the requests they want to import via a dialog. Upon selection, the corresponding requests are created in the API Dash collection within the workspace.

Key Changes:

 - [x] Added support for parsing HAR files.
 - [x] Implemented a dialog interface for users to select requests for import.
 - [x] Created functionality to generate corresponding requests in the API Dash collection.
 - [x] Respective Tests

<details>
<summary>Video Demo</summary>

https://github.com/foss42/apidash/assets/84124091/96364910-d260-4b7e-84a0-0f2b2a39b6bd

</details>

This PR lays the foundation for seamlessly importing API requests from HAR files, enhancing the usability and flexibility of the application. 

Tests will be added in subsequent commits to ensure the robustness and reliability of the newly introduced features.

Please review and provide feedback.

### CODE REVIEW

**Comment by opxdelwin** [comment]
Regarding the tests, I'm planning to create a dir `tests/dialogs/import` and test HAR implementation there. Looking for alternative suggestions...

File: test/widget_test.dart
Comment (animator) [inline test/widget_test.dart:1]: Revert changes in this file. It is for code coverage calculation purposes.

**Comment by animator** [comment]
Definitely a good start. I will review the PR and add my feedback.
Thanks!

File: test/widget_test.dart
Comment (opxdelwin) [reply test/widget_test.dart:1]: It's reverted, thanks!

**Comment by opxdelwin** [comment]
**Update**
Split business logic into separate files, to improve project structure.
This means easy integration of future import/export file features.

https://github.com/foss42/apidash/assets/84124091/85fef18f-9f9e-4d94-9d87-08bb106b777b

**Comment by animator** [comment]
On running the application and clicking Import, I am getting an error.
Platform: macOS

Screenshot 2024-03-15 at 8 36 40 PM

**Comment by opxdelwin** [comment]
Will have a look

**Comment by animator** [comment]
Any Updates?

**Comment by opxdelwin** [comment]
Sorry @animator I haven't been working lately as the semester end is approaching.

I'll be on limited work force till April 23. I'm looking to work on my open issues during gsoc community bonding period.

Feel free to ping me if anything arises!

**Comment by animator** [comment]
@opxdelwin no worries. Good luck for your exams!

**Comment by animator** [comment]
The architecture needs to be improved as there is currently no separation between presentation layer, services, utilities, etc. Also, there are already existing reported issues.
A new PR can be raised after the approach & design is discussed in detail on the issue thread.

**Pull request closed without merging**